src/rapids_dependency_file_generator/schema.json

```diff
@@ -162,22 +163,16 @@
         "extras": {
             "type": "object",
             "properties": {
+                "section": {
+                    "type": "string",
+                    "enum": ["build", "host", "run"]
+                },
                 "table": {
                     "type": "string",
                     "enum": ["build-system", "project", "project.optional-dependencies"]
                 },
-                "key": {"type": "string"}
-            },
-            "if": {
-                "properties": { "table": { "const": "project.optional-dependencies" } }
-            },
-            "then": {
-                "required": ["key"]
-            },
-            "else": {
-                "not": {
-                    "required": ["key"]
-                }
```

Comment:
CC @csadorf if you have thoughts here.

Reply:
So when you simply added a `"required": ["table"]` to the `"if"` block and removed the `"then"` and `"else"` blocks it would always require it? That seems indeed very odd. Can you provide some valid/invalid examples that we can test against?